compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1782,15 +1782,19 @@ object Types {
       else d.atSignature(sig).checkUnique
     }
 
-    override def newLikeThis(prefix: Type)(implicit ctx: Context): TermRef = {
-      val candidate = TermRef.withSig(prefix, name, sig)
+    private def fixDenot(candidate: TermRef, prefix: Type)(implicit ctx: Context): TermRef =
       if (symbol.exists && !candidate.symbol.exists) { // recompute from previous symbol
         val ownSym = symbol
         val newd = asMemberOf(prefix, allowPrivate = ownSym.is(Private))
```

Comment:
Should we revert the addition of the `allowPrivate` parameter to `asMemberOf` since it is no longer useful for ParamForwarder and could mask other bugs that only show up under `-Ytest-pickler` ?